dozer-ingestion/src/connectors/object_store/table_reader.rs

```diff
@@ -143,21 +144,48 @@ impl<T: DozerObjectStore> Reader<T> for TableReader<T> {
                 Arc::new(params.object_store),
             );
 
-            rt.block_on(Self::read(
-                id as u32,
-                ctx,
-                table_path,
-                listing_options,
-                ingestor,
-                table,
-            ))?;
```

Comment:
We shouldn't start many tokio runtimes. Every table read is an asynchronous task, one runtime should be enough to parallelize them.